src/components/dialog/dialog.spec.js

```diff
@@ -424,17 +417,13 @@ describe('Dialog', () => {
       });
 
       it('closes when the exit icon is click', () => {
-        wrapper.find('.carbon-dialog__close').at(0).simulate('click');
+        wrapper.find(IconButton).first().simulate('click');
         expect(onCancel).toHaveBeenCalled();
       });
 
       it('closes when exit icon is focused and Enter key is clicked', () => {
-        wrapper.find('.carbon-dialog__close').at(0).props().onKeyDown({ which: 13, preventDefault });
-        expect(onCancel).toHaveBeenCalled();
-      });
-
-      it('closes when exit icon is focused and Enter key is clicked', () => {
-        wrapper.find('.carbon-dialog__close').at(0).props().onKeyDown({ which: 16 });
+        const icon = wrapper.find(IconButton).first();
+        icon.simulate('keyDown', { which: 16 });
         expect(onCancel).not.toHaveBeenCalled();
```

Comment:
This test is wrong. There should be two cases. One for not enter, one for enter.

This test says "closes when exit icon is focused and Enter key is clicked" but uses `{ which: 16 }` which is `Shift`.

We should use `icon.simulate('keyDown', { key: 'Enter' });`

Reply:
Thanks Jamie! the `key: 'Enter'` alone doesn't seem to work, I added `which: 13, key: 'Enter'`.
665429c

Reply:
Ah, the `Event` helper mustn't have been updated yet. The change you've made allows for the existing version and the updated version to work 👍